src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Startup.cs

```diff
@@ -65,16 +61,36 @@ public void ConfigureServices(IServiceCollection services)
                 .AddIdentityServerJwt();
 #endif
 #if (OrganizationalAuth)
-#pragma warning disable CS0618 // Type or member is obsolete
-            services.AddAuthentication(AzureADDefaults.BearerAuthenticationScheme)
-                .AddAzureADBearer(options => Configuration.Bind("AzureAd", options));
-#pragma warning restore CS0618 // Type or member is obsolete
+            // Adds Microsoft Identity platform (AAD v2.0) support to protect this Api
+            services.AddMicrosoftWebApiAuthentication(Configuration, "AzureAd")
+#if (GenerateApiOrGraph)
+                    .AddMicrosoftWebApiCallsWebApi(Configuration,
+                                                   "AzureAd")
+                    .AddInMemoryTokenCaches();
+
+#else
+                    ;
```

Comment:
You need to put ` services.AddMicrosoftWebApiAuthentication(Configuration, "AzureAd")` inside the if and use ` services.AddMicrosoftWebApiAuthentication(Configuration, "AzureAd");` on the else.

Otherwise you generate 
```
services.AddMicrosoftWebApiAuthentication(Configuration, "AzureAd")
;
```

Reply:
I don't understand what you are saying. There are two #if, which one are you referring to. Also, I think the idea is to have just 
```
services.AddMicrosoftWebApiAuthentication(Configuration, "AzureAd")
;
```
if OrganizationAuth is configured but no API or Graph is configured? This is what's being done in the other templates. Is that wrong? cc @jmprieur